third_party/openthread/CMakeLists.txt

```
@@ -71,6 +71,9 @@ if (OT_THREAD_VERSION STREQUAL "1.2")
     set(OT_MLR ON CACHE BOOL "Enable Thread 1.2 MLR by default")
 endif()
 
+set(OT_COAP ON CACHE BOOL "Enable CoAP in OpenThread" FORCE)
```

It seems this disables us from disabling the feature even with `-DOT_COAP=OFF`.
```suggestion
set(OT_COAP ON CACHE BOOL "Enable CoAP in OpenThread")
```

---

Fixed! Thanks.